pkg/kubectl/cmd/cp.go

```diff
@@ -225,16 +227,14 @@ func copyFromPod(f cmdutil.Factory, cmd *cobra.Command, cmderr io.Writer, src, d
 		},
 
 		// TODO: Improve error messages by first testing if 'tar' is present in the container?
-		Command:  []string{"tar", "cf", "-", src.File},
+		Command:  []string{"tar", "cf", "-", prefix},
```

Comment:
This is not a good approach, since you're cutting off the initial `/` which might lead to getting a file you didn't expect to get. The warning tar gives is a normal functioning and should not be removed in this way. See https://unix.stackexchange.com/questions/59243/tar-removing-leading-from-member-names for more information.

Reply:
thank, will try and fix

Reply:
done， PTAL

Reply:
Did you push your changes? I can't see them.